dataflow/src/checktable/mod.rs

```diff
@@ -285,6 +285,27 @@ impl CheckTable {
         }
     }
 
+    pub fn recover(&mut self, base: NodeIndex) -> (i64, Option<Box<HashMap<domain::Index, i64>>>) {
```

Comment:
I'd like a comment over this function explaining exactly what it does.
I *think* what it does is reserve a single timestamp for the recovery, and then make it the latest timestamp for *every* base, essentially to bootstrap the timestamping logic. However, I don't know why it is *necessary*? Aren't the bases all synchronized at the beginning with `ts=0`? When the system crashes and comes back up, there are no writes anywhere in the graph, so taking a time seems unnecessary?

Reply:
It actually reserves a timestamp for the recovery of a specific base, and then marks that time as the latest timestamp for all columns in that base. This is necessary because at `ts=0` all base nodes are empty, and the recovery messages flowing through the graph are implemented as writes (and thus each must have some distinct timestamp assigned to them).